TeamCode/src/main/java/org/firstinspires/ftc/teamcode/KushaalDriverOpMode.java

```diff
@@ -110,47 +107,34 @@ public void runOpMode() throws InterruptedException {
 
         double speed = 0.75;
         double strafe = 0;
-
+        double max;
         // run until the end of the match (driver presses STOP)
         while (opModeIsActive()) {
-            double max;
 
-            if (speed < 0){
+            if (speed < 0) {
                 speed = 0;
             }
 
-            if (speed > 100){
+            if (speed > 100) {
                 speed = 100;
             }
 
-            if ((gamepad1.dpad_down) || (gamepad1.dpad_up)) {
-                while (gamepad1.dpad_down) {
-                    speed -= 25;
-                    while (gamepad1.dpad_down) {
-                    }
-                    speed += 25;
-                }
-                while (gamepad1.dpad_up) {
-                    speed += 25;
-                    while (gamepad1.dpad_up) {
-                    }
-                    speed -= 25;
-                }
-            }
-            if (gamepad1.dpad_left){
+            if (gamepad1.dpad_left) {
                 speed -= 25;
```

Comment:
Here, you can have `speed` < 0. Instead I think you should call the method `decrease_speed()`, which will cap the speed to 0.